### PR TITLE
tests added for thirdparty recipe

### PR DESCRIPTION
**Pull request by bhumilsarvaiya**
- basic tests added for third party recipe
- list of test todos added wherever required

### CODE REVIEW

**Comment by rishabhpoddar** [comment]
@bhumilsarvaiya good work on this so far. Please coordinate with @jscyo to complete the rest of the tests (whichever you think are important). Don't merge this branch until it's all done.

**Comment by rishabhpoddar** [comment]
@bhumilsarvaiya please review @jscyo's work here properly.

**Comment by jscyo** [comment]
Additional Tests to be added:
- test passing invalid private key in client secret when provider is apple, should throw bad input error
- test additional configs in config tests

**Comment by bhumilsarvaiya** [comment]
@jscyo Yes! Apart from those tests, all other changes/tests are correct. 👍🏼

File: test/thirdparty/provider.test.js [hunk 1]
```diff
@@ -704,4 +704,45 @@ describe(`providerTest: ${printPath("[test/thirdparty/provider.test.js]")}`, fun
...
+                            ThirParty.Apple({
+                                clientId,
+                                clientSecret,
+                            }),
+                        ],
+                    },
+                }),
+            ],
+        });
+        assert(false)
```
Comment (bhumilsarvaiya) [inline test/thirdparty/provider.test.js:741]: @jscyo Please correct the identation. make sure to run `npm run build-pretty`

**Pull request merged**